juniper/src/lib.rs

```diff
@@ -180,36 +179,38 @@ pub enum GraphQLError<'a> {
 }
 
 /// Execute a query in a provided schema
-pub fn execute<'a, CtxT, QueryT, MutationT>(
+pub fn execute<'a, S, CtxT, QueryT, MutationT>(
```

Comment:
It should here be easily possible to define a default type for the new generic type.

Something like:
```rust
pub fn execute<'a, S = DefaultScalarValue, CtxT, QueryT, MutationT>(
   document_source: &'a str,
    operation_name: Option<&str>,
    root_node: &'a RootNode<S, QueryT, MutationT>,
    variables: &Variables<S>,
    context: &CtxT,
) -> Result<(Value<S>, Vec<ExecutionError<S>>), GraphQLError<'a>>
where
    S: ScalarValue,
    for<'b> &'b S: ScalarRefValue<'b>,
    QueryT: GraphQLType<S, Context = CtxT>,
    MutationT: GraphQLType<S, Context = CtxT>,
```

I've not done that yet, because it is easier to debug/compile everything without that.

(Also maybe here the generic scalar type should be the last one in the list)

Reply:
So I was wrong. It is not possible to define default types for functions, only for types.